frontend/src/_components/CreateApp.jsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState, useEffect, useContext } from 'react';
+import { toast } from 'react-hot-toast';
+import Modal from '../HomePage/Modal';
+import { ButtonSolid } from '@/_ui/AppButton/AppButton';
+import { appService } from '@/_services';
+import _ from 'lodash';
+
+export function CreateApp({ closeModal, createApp, show }) {
+  const [appName, setappName] = useState('');
+  const [errorText, setErrorText] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleCreateApp = async () => {
+    if (!errorText) {
+      setIsLoading(true);
+      const appExists = await checkIfAppExists(appName);
+      if (appExists) {
+        setErrorText('App name already exists');
+      } else {
+        createApp(appName);
+        closeModal();
+      }
+      setIsLoading(false);
+    }
+  };
+
+  const checkIfAppExists = async (appName) => {
+    try {
+      const response = await appService.getAll(1, null, appName);
```

Comment:
Can we avoid this extra API call?
might take more time for a workspace which has a large quantity of apps

One thing you can think of,

Add an API-level check for unique names. So instead of 
`const appExists = await checkIfAppExists(appName);`. Try to return the error messages from `createApp()` fn and do `setErrorText()`